Cargo.toml

```diff
@@ -20,4 +20,6 @@ include = [
 
 [dev-dependencies]
 lazy_static = "1.0"
+maplit = "1"
+pretty_assertions = "0.5"
```

Comment:
Is it common practice to leave out "trailing" zeros? It would seem more consistent to provide the full crate IDs.

Reply:
Yes, this is common practice.

Something like

~~~
foo = "1.2.0"
~~~

Actually means versions >= 1.2.0 and < 2.0.0. So, specifying 1.2.0 and 1.2 are equivalent. Besides that, Rust/Cargo mandates [semver](https://semver.org). In semantic versioning, there may not be any API-breaking changes within a major version. So, you can safely put

~~~
foo = "1"
~~~

which means versions >= 1.0.0 and < 2.0.0, since any 1.x.y must be API compatible. It is permitted to add new features (APIs) in some *x* version (e.g. 1.3.0). So, would say something like:

~~~
foo = "1.3"
~~~

if you project depends on an API that was only added in version. But if you do not use any new API's you should just use version "1", since that makes it easier for the dependency solver to solve the dependency constraints.

---

One big exception to all of this is major version 0. No API compat is expected for different values of x for 0.x. Therefore, cargo treats e.g. 0.5 as >= 0.5.0 and < 0.6.0. For this reason, 0.x package versions are more fully specified.

---

Yep, the lazy-static dependency should also be just 1 ;).